### ISSUE TITLE
Gateway Error?

### ISSUE THREAD

**Issue by satheevrin** [rigormorrtiss/discordo#346]
Here are my logs: 
<code>2024/02/24 09:48:52 ws.go:16: Gateway error: websocket: close sent
2024/02/24 09:48:52 main.go:22: websocket closed, reason: 4004 Authentication failed.</code>

I tried building this from source, thinking that I might be able to get some progress, but no luck. Am I missing something? A dependency?

Would anyone please explain to me what the log is telling me??

**Comment by sewnie** [comment]
Have you tried searching up any available issues? This is a duplicate of #342. You may need an authentication code.

**Comment by ayn2op** [comment]
> Here are my logs: `2024/02/24 09:48:52 ws.go:16: Gateway error: websocket: close sent 2024/02/24 09:48:52 main.go:22: websocket closed, reason: 4004 Authentication failed.`
> 
> I tried building this from source, thinking that I might be able to get some progress, but no luck. Am I missing something? A dependency?
> 
> Would anyone please explain to me what the log is telling me??

If your account has MFA enabled, input the MFA code in the "Code" input field and continue.

**Comment by satheevrin** [comment]
My account does not have MFA enabled. I see what you're saying, but I have no code to input for this line.

EDIT: I can login using the token method with the flag, but not the normal method.

EDIT: I have some more troubleshooting stuff: Interestingly enough, this works on a VM on a fresh arch install with no problems or arguments needed. IE: The login screen will show and the program will work as it is intended without the token method.  My install of Arch (Which isn't a VM) is fresh as well. What could be going on here? Why's mine not behaving like it should in the VM??

EDIT: Ahh, I see. Okay, so I deleted my keyrings folder in /.share . This brought the login screen back, but unfortunately, it does not save my login anymore. Any tips?

EDIT: Okay, looks like I fixed it!! To summarize: I deleted the keyrings folder, started Discordo up with no arguments or tokens. It takes me to the login screen where I enter my credentials. I'm now in the terminal with all my servers, etc. At this point, DON'T close it by ctrl-c, just exit it out. It should work normally from now on.

Let me know if it works! Seems my problem at least could have been because of a bad keyring, which I copied from my last machine.
